pkg/bitwarden/cli.go

```diff
@@ -142,3 +151,246 @@ func (c *cliClient) GetPassword(itemName string) ([]byte, error) {
+	}
+	return c.createItem(string(itemBytes), targetItem)
+}
+
+func (c *cliClient) editItem(targetItem Item) error {
+	targetJSON, err := json.Marshal(targetItem)
+	if err != nil {
+		return fmt.Errorf("failed to marshal object: %w", err)
+	}
+	encodedItem := base64.StdEncoding.EncodeToString(targetJSON)
+	_, err = c.runWithSession("edit", "item", targetItem.ID, encodedItem)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (c *cliClient) deleteAttachment(attachmentID, itemID string) error {
+	if _, err := c.runWithSession("delete", "attachment", attachmentID, "--itemid", itemID); err != nil {
+		return fmt.Errorf("failed to delete attachment, attachmentID: %s, itemID: %s: %w", attachmentID, itemID, err)
+	}
+	return nil
+}
+
+func (c *cliClient) SetFieldOnItem(itemName, fieldName string, fieldValue []byte) error {
+	var targetItem *Item
+	var targetField *Field
+	for index, item := range c.savedItems {
+		if itemName == item.Name {
+			targetItem = &c.savedItems[index]
+			for fieldIndex, field := range item.Fields {
+				if field.Name == fieldName {
+					targetField = &c.savedItems[index].Fields[fieldIndex]
+					break
+				}
+			}
+			break
+		}
+	}
+	if targetItem == nil {
+		newItem := &Item{}
+		if err := c.createEmptyItem(itemName, newItem); err != nil {
+			return fmt.Errorf("failed to create new	 bw entry: %w", err)
+		}
+		c.savedItems = append(c.savedItems, *newItem)
+		targetItem = &c.savedItems[len(c.savedItems)-1]
+	}
+	isNewField := false
+	if targetField == nil {
+		targetItem.Fields = append(targetItem.Fields, Field{fieldName, string(fieldValue)})
+		targetField = &targetItem.Fields[len(targetItem.Fields)-1]
+		isNewField = true
+	}
+	if isNewField || targetField.Value != string(fieldValue) {
+		targetField.Value = string(fieldValue)
+		if err := c.editItem(*targetItem); err != nil {
+			return fmt.Errorf("failed to set field, itemName: %s, fieldName: %s - %w", itemName, fieldName, err)
+		}
+	}
+	return nil
+}
+
+func (c *cliClient) SetAttachmentOnItem(itemName, attachmentName string, fileContents []byte) error {
+	var targetItem *Item
+	var targetAttachment *Attachment
+	var targetAttachmentIndex int
+	for index, item := range c.savedItems {
+		if itemName != item.Name {
+			continue
+		}
+		targetItem = &c.savedItems[index]
+		for attachmentIndex, attachment := range item.Attachments {
+			if attachment.FileName == attachmentName {
+				targetAttachmentIndex = attachmentIndex
+				targetAttachment = &c.savedItems[index].Attachments[attachmentIndex]
+				break
+			}
+		}
+		break
+
+	}
+	if targetItem == nil {
+		newItem := &Item{}
+		if err := c.createEmptyItem(itemName, newItem); err != nil {
+			return fmt.Errorf("failed to create new	 bw entry: %w", err)
+		}
+		c.savedItems = append(c.savedItems, *newItem)
+		targetItem = &c.savedItems[len(c.savedItems)-1]
+	}
+	attachmentChanged := true
+	if targetAttachment != nil {
+		// read the attachment file
+		tempDir, err := ioutil.TempDir("", "attachment")
```

Comment:
Why create a tempdir per attachment rather than one per client lifecycle?

Reply:
Thats for if we end up making this concurrent, then we would need unique locations for each of the attachment files and avoid conflicts with similar attachment names

Reply:
But right now its not concurrent so its a premature optimization. I would except one base tempdir with a subdir per item and a file per field/property/attachment in which case this wouldn't be needed

Reply:
I would need either a temp directory or temp file for holding the attachment file to pass as a parameter to bw create. If the user wants to see the output of the commands the user is expected to run it in dry run mode which would not create a temp-directory. The indentation in the output file would give the user an idea of the itemname-field/attachment /attribute organization in bitwarden